src/components/InventoryV2/form/BatchDetailsForm.tsx

```diff
@@ -230,6 +230,24 @@ export default function BatchDetailsForm(props: BatchDetailsModalProps): JSX.Ele
     }
   }, [record, selectedAccession, setRecord]);
 
+  // when sublocations change, clear current sublocations in the record
+  useEffect(() => {
+    // if this is a read-only view of an existing batch, don't update the sublocations
+    if (record?.id !== -1) {
+      return;
+    }
+    setRecord((previousRecord: FormRecord): FormRecord => {
+      if (!previousRecord) {
+        return previousRecord;
+      }
+
+      return {
+        ...previousRecord,
+        subLocationIds: [],
+      };
+    });
+  }, [availableSubLocations, record?.id, setRecord]);
+
   return (
     <>
       {record && (
```

Comment:
@nickgraz would you know how I can trigger this read-only view? can't seem to find the use-case in the inventory views.

Reply:
It used to be the main way to view a batch's details, but now they are forwarded to the batch details page. I think you can still trigger it if you have a q

Reply:
It used to be the main way to view a batch's details, but now they are forwarded to the batch details page. I think you can still trigger it if you have a query param in the parent view for the open batch. I suspect there are still other views out there that might link to it that way.